modules/obstacle.py

```diff
@@ -0,0 +1,108 @@
+"""
+Obstacle data structure.
+"""
+
+from . import lidar_detection
+
+
+class Obstacle:
+    """
+    Contains obstacles that represent lidar readings.
+
+    The following are valid obstacle contructors:
+
+    * Obstacle.create_circle_obstacle
+    * Obstacle.create_line_obstacle
+    * Obstacle.create_rect_obstacle
+    """
+
+    create_key = object()
+
+    @classmethod
+    def create_circle_obstacle(
+        cls, centre: lidar_detection.LidarDetection, radius: float
+    ) -> "tuple[bool, Obstacle.Circle | None]":
+        """
+        Circle obstacle contructor.
+        """
+        return True, Obstacle(cls.create_key, centre, radius)
```

Comment:
check radius > 0, add to docstring explaining what the centre is - does it represent coordinates?